doc/cpp-auto-tool-architecture-documentation.md

```diff
@@ -0,0 +1,256 @@
+
+```json
+{
+  "PackageName": "inireader",
+  "HeaderFiles": [
+    "/path/to/header/INIReader.h",
+    "/path/to/header/AnotherHeader.h"
+  ],
+  "DLLFile": "/path/to/lib/libINIReader.dylib",
+  "JSONFile": "/path/to/json/config.json"
+}
+```
+
+```bash
+./generate_symbol_table /path/to/config.json
+```
+
+### Implementation Steps
+
+1. Parse dylib and store:
+
+```go
+// common.go
+type CPPSymbol struct {
+    Address string
+    Type    string
+    Name    string
+}
+
+// parser_dylib.go
+func ParseDylibSymbols(dylibPath string) ([]common.CPPSymbol, error)
+```
+
+2. Parse header files and store:
+
+```go
+// common.go
+type ASTInformation struct {
+    Namespace   string
+    Class       string
+    Name        string
+    BaseClasses []string
+    ReturnType  string
+    Location    string
+    Parameters  []Parameter
+    Symbol      string
+}
+
+type Parameter struct {
+    Name string
+    Type string
+}
+
+// parser_ast.go
+func ParseHeaderFile(files []string) ([]common.ASTInformation, error)
+```
+
+3. Cross-reference data from the first two steps to get the final output
+
+```go
+// common.go
+type CommonSymbolInfo struct {
+    FunctionName       string
+    Symbol             string
+    Location           string
+    UserFunctionName   string
+}
+
+// common_symbols.go
+func GetCommonSymbols(dylibSymbols []common.CPPSymbol, astInfoList []common.ASTInformation) []common.CommonSymbolInfo
+```
+
+4. Generate `common_symbol_info.json` file and store the JSON file path into `config.json`
+
+```go
+// generator.go
+func GenerateJSON([]CommonSymbolInfo)
+```
+
+5. Example `common_symbol_info.json` file
+
+```json
+{
+    "FunctionName": "A::B::C",
+    "Symbol": "_ZN9INIReaderC1ERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEE",
+    "Location": "a.h",
+    "UserFunctionName": "CFromA"
+}
+```
+
+## Function Declaration Generation Module
+
+### Input
+
+No input required, directly reads the `config.json` file
+
+### Implementation Steps
+
+1. Execute the executable
+
+```bash
+./generate_func_decl path/to/config.json
+```
+
+2. Parse header files
+
+```go
+// common.go
+type ASTInformation struct {
+    Namespace   string
+    Class       string
+    Name        string
+    BaseClasses []string
+    ReturnType  string
+    Location    string
+    Parameters  []Parameter
+}
+
+type Parameter struct {
+    Name string
+    Type string
+}
+
+// parser_ast.go
+func ParseHeaderFile(filePath string) ([]common.AstInformation, error)
+```
+
+3. Generate the final JSON mapping file `func_prototype.json`
+
+```go
+ func GenerateJSONFile(info []common.AstInformation)
+```
+
+```json
+{
+    "FunctionName": "A::B::C",
+    "Symbol": "_ZN9INIReaderC1ERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEE",
+    "Location": "a.h",
+    "ReturnType" : "int",
+    "UserFunctionName": "CFromA",
+    "Parameters" : [
+        {
+            "arg1" : "int",
+        },
```

Comment:
A trailing comma appears after the last key-value pair in this JSON example, which should be removed to ensure consistent trailing comma style.
```suggestion
        {
            "arg1" : "int"
        },
```